site/src/pages/WorkspacePage/ResourcesSidebar.tsx

```diff
@@ -0,0 +1,109 @@
+import { Interpolation, Theme } from "@emotion/react";
+import Skeleton from "@mui/material/Skeleton";
+import { useTheme } from "@mui/material/styles";
+import { WorkspaceResource } from "api/typesGenerated";
+import {
+  Sidebar,
+  SidebarCaption,
+  SidebarItem,
+} from "components/FullPageLayout/Sidebar";
+import { getResourceIconPath } from "utils/workspace";
+
+type ResourcesSidebarProps = {
+  failed: boolean;
+  resources: WorkspaceResource[];
+  onChange: (resourceId: string) => void;
+  selected: string;
+};
+
+export const ResourcesSidebar = (props: ResourcesSidebarProps) => {
+  const theme = useTheme();
+  const { failed, onChange, selected, resources } = props;
+
+  return (
+    <Sidebar>
+      <SidebarCaption>Resources</SidebarCaption>
+      {failed && (
+        <p
+          css={{
+            margin: 0,
+            padding: "0 16px",
+            fontSize: 13,
+            color: theme.palette.text.secondary,
+            lineHeight: "1.5",
+          }}
+        >
+          Your workspace build failed, so the necessary resources couldn&apos;t
+          be created.
+        </p>
+      )}
+      {resources.length === 0 &&
+        !failed &&
+        Array.from({ length: 8 }, (_, i) => (
+          <SidebarItem key={i}>
+            <ResourceSidebarItemSkeleton />
+          </SidebarItem>
+        ))}
+      {resources.map((r) => (
+        <SidebarItem
+          onClick={() => onChange(r.id)}
+          isActive={r.id === selected}
+          key={r.id}
+          css={styles.root}
+        >
+          <div
+            css={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              lineHeight: 0,
+              width: 16,
+              height: 16,
+              padding: 2,
+            }}
+          >
+            <img
+              css={{ width: "100%", height: "100%", objectFit: "contain" }}
+              src={getResourceIconPath(r.type)}
+              alt=""
+            />
+          </div>
+          <div
+            css={{ display: "flex", flexDirection: "column", fontWeight: 500 }}
+          >
+            <span>{r.name}</span>
+            <span css={{ fontSize: 12, color: theme.palette.text.secondary }}>
+              {r.type}
+            </span>
+          </div>
+        </SidebarItem>
+      ))}
+    </Sidebar>
+  );
+};
+
+export const ResourceSidebarItemSkeleton = () => {
+  return (
+    <div css={[styles.root, { pointerEvents: "none" }]}>
```

Comment:
Does `pointerEvents` do anything here? Divs aren't interactive by default, and turning on the style rule would just make clicks on the div hit the parent directly, which wouldn't be _that_ different from the events firing on the div, and then bubbling up to the parent